mirdata/datasets/baf.py

```diff
@@ -0,0 +1,436 @@
+import librosa
+import numpy as np
+import pandas as pd
+
+from mirdata import annotations
+from mirdata import core
+from mirdata import jams_utils
+
+
+BIBTEX = """@inproceedings{cortes2022BAF,
+  author       = {Guillem Cortès and
+                  Alex Ciurana and
+                  Emilio Molina and
+                  Marius Miron and
+                  Owen Meyers and
+                  Joren Six and
+                  Xavier Serra},
+  title        = {BAF: An audio fingerprinting dataset for broadcast monitoring},
+  booktitle    = {Proceedings of the 23rd International Society for Music Information Retrieval Conference},
+  year         = 2022,
+  pages        = {908-916},
+  publisher    = {ISMIR},
+  address      = {Bengaluru, India},
+  month        = dec,
+  venue        = {Bengaluru, India},
+  doi          = {10.5281/zenodo.7316812},
+  url          = {https://doi.org/10.5281/zenodo.7372162}
+}
+"""
+
+INDEXES = {
+    "default": "1.0",
+    "test": "1.0",
+    "1.0": core.Index(filename="baf_index_1.0.json"),
+}
+
+REMOTES = None
+
+DOWNLOAD_INFO = (
+    "BAF dataset is only available upon request. To download the audio "
+    "request access in this link: https://doi.org/10.5281/zenodo.6868083. "
+    "Then unzip the audio into the baf general dataset folder for the rest of "
+    "annotations and files. Please include, in the justification field, your "
+    "academic affiliation (if you have one) and a brief description of your "
+    "research topics and why you would like to use this dataset.\n"
+    "    baf/\n"
+    "    ├── baf_datasheet.pdf\n"
+    "    ├── annotations.csv\n"
+    "    ├── changelog.md\n"
+    "    ├── cross_annotations.csv\n"
+    "    ├── queries_info.csv\n"
+    "    ├── queries\n"
+    "    │   ├── query_0001.wav\n"
+    "    │   ├── query_0002.wav\n"
+    "    │   ├── …\n"
+    "    │   └── query_3425.wav\n"
+    "    ├── queries_info.csv\n"
+    "    └── references\n"
+    "        ├── ref_0001.wav\n"
+    "        ├── ref_0002.wav\n"
+    "        ├── …\n"
+    "        └── ref_2000.wav\n"
+)
+
+LICENSE_INFO = (
+    "Given the different ownership of the elements of the dataset, the "
+    "dataset is licensed under the following conditions:\n"
+    "    * User's access request\n"
+    "    * Research only, non-commercial purposes\n"
+    "    * No adaptations nor derivative works\n"
+    "    * Attribution to Epidemic Sound and the authors as it is indicated "
+    "in the ”citation” section.\n"
+)
+
+#: Tag units
+TAG_UNITS = {"open": "no scrict schema or units"}
+
+FILENOTFOUND_MSG = Template(
+    "$fname not found. Check that the file is found in the dataset root "
+    "directory e.g. mir-datasets/baf/$fname"
+)
+
+
+@core.docstring_inherit(annotations.EventData)
+class EventDataExtended(annotations.EventData):
+    """EventDataExtended class
+
+    Attributes:
+        tags (list): list of tag labels (as strings)
+        tag_unit (str): tag units, one of TAG_UNITS
+
+    """
```

Comment:
Looks good, and I personally like the idea of this class here. However, I think it'd be good to briefly present in the docstring what is the difference between the original `EventData` and the extended version here! Just in a single sentence would be enough :) 
That class is a nice example for the issue in #558.